### PR TITLE
feat(util/Text): support soft breaks

### PR DESCRIPTION
**Pull request by nikku**
This adds handling for discretionary hyphens / soft breaks in our renderer.

If a discretionary hyphen is rendered as a break point it becomes a `-`.

Related to https://github.com/bpmn-io/bpmn-js/issues/1383

---

capture KCFgUj_optimized

### CODE REVIEW

File: lib/util/Text.js
Comment (azeghers) [inline lib/util/Text.js:138]: The regex doesn't look like it contains a special character at the end, it could inadvertently be removed by a future maintainer. I think this line deserves a comment that warns that a soft hyphen is present in the regex.
Or alternatively, you can match it using the unicode value `\u00AD`.

Comment (nikku) [reply lib/util/Text.js:138]: Agreed. Good catch.

Comment (nikku) [reply lib/util/Text.js]: Using explicit unicode char now. Much better: https://github.com/bpmn-io/diagram-js/pull/508/commits/3981c21ae4c79ca47526636376cd1c6e13f84774

**Comment by azeghers** [comment]
LGTM
Commits should be squashed before merging, right?

**Review by azeghers — APPROVED** [review]

**Pull request merged**
